javascript/cw_tool.path.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * @file
+ *  Add path related behaviours.
+ */
+var CW = CW || {};
+
+(function($) {
+
+  'use strict';
+    
+  CW.Path = {
```

Comment:
For now I will add _ at the beginning of the private methods.
But in the future this pattern is better:
https://www.youtube.com/watch?v=DwYPG6vreJg
http://stackoverflow.com/questions/1813113/how-to-inherit-a-private-member-in-javascript